src/utilities.jl

```diff
@@ -171,19 +171,19 @@ Create a raster from the scattered data.
 The output dataset.
 """
 function unsafe_gdalgrid(
-        dataset::Dataset,
+        dataset::T,
         options = String[];
         dest = "/vsimem/tmp"
-    )
+    ) where T<:AbstractDataset
     options = GDAL.gdalgridoptionsnew(options, C_NULL)
     usage_error = Ref{Cint}()
     result = GDAL.gdalgrid(dest, dataset.ptr, options, usage_error)
     GDAL.gdalgridoptionsfree(options)
-    return Dataset(result)
+    return T(result)
```

Comment:
This should still be `Dataset(result)` rather than `T(result)` (see https://github.com/yeesian/ArchGDAL.jl/pull/167#issuecomment-765704084 for context).